source/mainwindow.cpp

```diff
@@ -542,7 +549,38 @@ void MainWindow::saveFile(SaveAsParam *params)
     this->ui->statusBar->showMessage("Saving...");
     this->ui->statusBar->repaint();
 
-    bool change = this->cel->save(params);
+    bool change = false;
+    QString filePath;
+    if (params != nullptr)
+        filePath = params->celFilePath;
+    if (filePath.isEmpty())
+        filePath = this->gfx->getFilePath();
+    if (this->gfx->getType() == D1CEL_TYPE::V1_LEVEL) {
+        if (!filePath.toLower().endsWith("cel")) {
+            QMessageBox::StandardButton reply;
+            reply = QMessageBox::question(nullptr, "Confirmation", "Are you sure you want to save as " + filePath + "? Data conversion is not supported.", QMessageBox::Yes | QMessageBox::No);
+            if (reply != QMessageBox::Yes) {
+                // Clear loading message from status bar
+                this->ui->statusBar->clearMessage();
+                return;
+            }
+        }
```

Comment:
I would probably just flat out refuse with a message like "Unsupported format for tilesets, file must end with .cel". If someone is dead set on mislabeling there cel files they can rename them afterwards.

Reply:
If they want to name it .bkp or .tmp, I'm not going to stop them. I would like to extend the 'Open As' dialog with the min/til/sol path parameters similar to the 'Save As' dialog.

Reply:
> Is the thinking here to allow for engines that support using cl2 as a tile graphics format?

I'm not sure the extension of tilesets matters that much, because the format is defined by the frame types (triangle, square, etc). There is nothing CEL-like encoding in there.